### PR TITLE
allow function replacement

### PR DESCRIPTION
**Pull request by arnej27959**
* for backwards-compatibility, allow a rank-profile function
  to be replaced (with a warning)
* also, allow multiple attribute settings for same attribute

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@bratseth or @baldersheim please review

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]

File: config-model/src/main/javacc/IntermediateParser.jj [hunk 1]
```diff
@@ -1910,7 +1907,12 @@ void function(ParsedRankProfile profile) :
     {
         func.setExpression(expression);
         func.setInline(inline);
-        profile.addFunction(func);
+        var old = profile.addOrReplaceFunction(func);
+        if (old != null) {
+            deployLogger.logApplicationPackage(Level.WARNING, "Function '" + func.name()
+                                               + "' replaces a previous function with the same name in rank profile '"
```
Comment (bratseth) [inline config-model/src/main/javacc/IntermediateParser.jj:1913]: Shouldn't this just say "[function] is defined twice"?

File: config-model/src/main/javacc/IntermediateParser.jj [hunk 2]
```diff
@@ -1910,7 +1907,12 @@ void function(ParsedRankProfile profile) :
     {
         func.setExpression(expression);
         func.setInline(inline);
-        profile.addFunction(func);
+        var old = profile.addOrReplaceFunction(func);
+        if (old != null) {
+            deployLogger.logApplicationPackage(Level.WARNING, "Function '" + func.name()
```
Comment (bratseth) [inline config-model/src/main/javacc/IntermediateParser.jj:1912]: I suggest a TODO on changing this to an error on Vespa 8.

File: config-model/src/main/javacc/IntermediateParser.jj (hunk 1 above)
Comment (arnej27959) [reply config-model/src/main/javacc/IntermediateParser.jj:1913]: > Shouldn't this just say "[function] is defined twice"?

I will change this (the unit test was very strict)

**Review by bratseth — APPROVED** [review]
👍

**Pull request merged**